ecpy/tasks/tasks/util/definition_task.py

```diff
@@ -0,0 +1,66 @@
+# -*- coding: utf-8 -*-
+# -----------------------------------------------------------------------------
+# Copyright 2015 by Ecpy Authors, see AUTHORS for more details.
+#
+# Distributed under the terms of the BSD license.
+#
+# The full license is in the file LICENCE, distributed with this software.
+# -----------------------------------------------------------------------------
+"""Task for defining various definitions.
+
+"""
+from __future__ import (division, unicode_literals, print_function,
+                        absolute_import)
+
+from collections import OrderedDict
+from traceback import format_exc
+
+from atom.api import Typed
+
+from ....utils.atom_util import (ordered_dict_from_pref, ordered_dict_to_pref)
+
+from ...base_tasks import SimpleTask
+
+
+class DefinitionTask(SimpleTask):
+    """Task defining a list of global variables in the database. Possibly used
+            in other tasks.
+    """
+    #: Class attribute marking this task as being logical, used in filtering.
+    util_task = True
+
+    # Dictionary of definitions
+    definitions = Typed(OrderedDict, ()).tag(pref=[ordered_dict_to_pref,
+                                                   ordered_dict_from_pref])
+
+    def perform(self):
+        """ Update the database.
+        """
+        for k, v in self.definitions.items():
+            value = self.format_string(v)
```

Comment:
This is not what we expect, definitions should evaluate the provided values but without performing any replacement. Please fix it and update the checks and tests accordingly. Use the safe_eval found in tools.string_evaluation